src/gg.regression.unity.bots/Editor/Scripts/BotSegments/IRGBotSequencesFinder.cs

```diff
@@ -58,21 +58,21 @@ private static void CreateBotSegmentsAsset(object o)
                     var fileContents = fr.ReadToEnd();
                     try
                     {
-                        var sequence = JsonConvert.DeserializeObject<BotSequence>(fileContents);
+                        var sequence = JsonConvert.DeserializeObject<BotSequence>(fileContents, JsonUtils.JsonSerializerSettings);
                         botSequences.sequences.Add(outputPath);
                     }
                     catch (Exception)
                     {
                         try
                         {
-                            var segmentList = JsonConvert.DeserializeObject<BotSegmentList>(fileContents);
+                            var segmentList = JsonConvert.DeserializeObject<BotSegmentList>(fileContents, JsonUtils.JsonSerializerSettings);
                             botSequences.segmentLists.Add(outputPath);
                         }
-                        catch (Exception)
+                        catch (Exception ex)
                         {
                             try
                             {
-                                var segment = JsonConvert.DeserializeObject<BotSegment>(fileContents);
+                                var segment = JsonConvert.DeserializeObject<BotSegment>(fileContents, JsonUtils.JsonSerializerSettings);
                                 botSequences.segments.Add(outputPath);
                             }
                             catch (Exception)
```

Comment:
(nit) It would be nice if we could tie `AssetDatabase.Refresh();` after copying.

However, I saw that we already do a refresh just below here, and I couldn't find a better place in the context of this PR. Typing this in case there is another possible refresh point that I missed.

Reply:
Copying a sequence happens while the game is running and thus AssetDatabase.Refresh() would only work as an API when running in the editor.  Even in the Editor though, there isn't a need to update the database because we aren't loading files from it.

In the editor runtime, we load the files as files never looking at the generated asset from the code above.

In the production runtime;  we load the asset as a 'resource' as well as loading the other in game segment files as 'resources' and overrides from the filesystem as files.

Ultimately this sequence file is only updated at compile time because it only impacts non-editor runtimes.

And.. in those runtimes, copying a sequence cannot update game binaries (writes to disk) so there wouldn't be a need to update this asset file even if it were possible there.

Reply:
Thank you for the explanation!